### PR TITLE
Button: add new transparentWhite color

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Added color to match color='white' in `IconButton`.

Kapture 2021-05-13 at 19 23 42

### Links

- JIRA: https://jira.pinadmin.com/browse/PDS-XXX
- [TDD](paper doc link)
- [Figma](link to figma file)

### Checklist

- [ ] Added Unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified Accessibility: keyboard & screen reader interaction
- [ ] Checked Dark Mode, Responsiveness, and Right-to-Left support
- [ ] Checked Stakeholder feedback (ex. Gestalt designers)

### CODE REVIEW

**Comment by ashleyseto** [comment]
I'm having trouble finding the example in the docs. Only a coupe notes for the example.

- It'd be good if we could put these over an image since our guidance will be to use over images
- We generally recommend the icon in the icon button be the dark gray rather than gray so we should show that here.

Thanks for getting this done so quickly!

File: docs/src/Button.doc.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ card(
       },
       {
         name: 'color',
-        type: `'gray' | 'red' | 'blue' | 'transparent' | 'transparentWhiteText' | 'white'`,
+        type: `'gray' | 'red' | 'blue' | 'transparent' |  'transparentWhite' | 'transparentWhiteText' | 'white'`,
```
Comment (ayeshakmaz) [inline docs/src/Button.doc.js:74]: This needs to be updated

File: docs/src/Button.doc.js [hunk 2]
```diff
@@ -327,48 +327,86 @@ card(
 );
 
 card(
-  <Combination id="color" name="Color" color={['gray', 'red', 'blue', 'white', 'transparent']}>
+  <Combination
+    id="color"
+    name="Color"
+    color={['gray', 'red', 'blue', 'white', 'transparent', 'transparentWhite']}
```
Comment (ayeshakmaz) [inline docs/src/Button.doc.js:333]: Same here

File: docs/src/Button.doc.js [hunk 3]
```diff
@@ -327,48 +327,86 @@ card(
...
+  >
     {(props, i) => <Button id={`example-${i}`} onChange={() => {}} {...props} text="Button" />}
   </Combination>,
 );
 
 card(
   <Example
-    name="Color: transparent"
+    name="Color: transparent and semiTransparentWhite"
```
Comment (ayeshakmaz) [inline docs/src/Button.doc.js:341]: Love the dual example!

File: packages/gestalt/src/Button.js [hunk 4]
```diff
@@ -205,7 +213,7 @@ const ButtonWithForwardRef: React$AbstractComponent<unionProps, unionRefs> = for
         wrappedComponent="button"
       >
         {iconEnd ? (
-          <IconEnd text={buttonText} textColor={textColor} icon={iconEnd} size={size} />
+          <IconEnd text={buttonText} textColor="darkGray" icon={iconEnd} size={size} />
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Button.js:216]: I don't think these two should be hard coded, it creates issues when you disabled or select them
Screen Shot 2021-05-14 at 9 48 35 AM

Was there a need to change them?

File: docs/src/Button.doc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:74]: DONE

File: docs/src/Button.doc.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:333]: DONE

File: packages/gestalt/src/Button.js (hunk 4 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js]: LEGACY DEBUGGING... MY BAD

DONE

**Review by ayeshakmaz — APPROVED** [review]
LGTM 👍

**Pull request merged**